modules.d/50plymouth/module-setup.sh

```diff
@@ -1,27 +1,77 @@
 #!/bin/bash
 
 pkglib_dir() {
-    local _dirs="/usr/lib/plymouth /usr/libexec/plymouth/"
+    local _dirs="/usr/lib/plymouth /usr/lib64/plymouth /usr/libexec/plymouth"
     if find_binary dpkg-architecture &> /dev/null; then
         local _arch
         _arch=$(dpkg-architecture -qDEB_HOST_MULTIARCH 2> /dev/null)
         [ -n "$_arch" ] && _dirs+=" /usr/lib/$_arch/plymouth"
     fi
     for _dir in $_dirs; do
-        if [ -x "$dracutsysrootdir""$_dir"/plymouth-populate-initrd ]; then
+        if [[ -x "$dracutsysrootdir""$_dir"/plymouthd-fd-escrow ]]; then
+            echo "$_dir"
+            return
+        fi
+        if [[ -x "$dracutsysrootdir""$_dir"/plymouth-populate-initrd ]]; then
+            echo "$_dir"
+            return
+        fi
+    done
+}
+
+plugin_dir() {
+    local _dirs="/usr/lib/plymouth /usr/lib64/plymouth /usr/libexec/plymouth"
+    if find_binary dpkg-architecture &> /dev/null; then
+        local _arch
+        _arch=$(dpkg-architecture -qDEB_HOST_MULTIARCH 2> /dev/null)
+        [[ -n $_arch ]] && _dirs+=" /usr/lib/$_arch/plymouth"
+    fi
+    for _dir in $_dirs; do
+        if [[ -d "$dracutsysrootdir""$_dir"/renderers ]]; then
             echo "$_dir"
             return
         fi
     done
 }
 
+# $1: Configuration file
+# $2: Setting name
+function read_setting_from_file() {
+    grep -v '^#' "$1" 2> /dev/null \
+        | awk 'BEGIN { FS="[=[:space:]]+"; OFS=""; ORS="" } $1 ~/^'"$2"'$/ { print $2 }'
+}
+
+# $1: Configuration file
+function set_configured_theme_path_from_file() {
+    [[ -z $CONFIGURED_THEME_DIR ]] && CONFIGURED_THEME_DIR=$(read_setting_from_file "$1" "ThemeDir")
+    [[ -z $CONFIGURED_THEME_DIR ]] && CONFIGURED_THEME_DIR=/usr/share/plymouth/themes
+    CONFIGURED_THEME_NAME=$(read_setting_from_file "$1" "Theme")
+}
+
+function set_theme_dir() {
+    set_configured_theme_path_from_file "$dracutsysrootdir"/etc/plymouth/plymouthd.conf
+    if [[ -z $CONFIGURED_THEME_DIR ]] || [[ ! -d "$dracutsysrootdir$CONFIGURED_THEME_DIR/$CONFIGURED_THEME_NAME" ]]; then
+        set_configured_theme_path_from_file "$dracutsysrootdir"/usr/share/plymouth/plymouthd.defaults
+    fi
+
+    if [[ -n $CONFIGURED_THEME_DIR ]] && [[ -d $dracutsysrootdir$CONFIGURED_THEME_DIR ]]; then
+        PLYMOUTH_THEME_DIR="$CONFIGURED_THEME_DIR"
+    else
+        PLYMOUTH_THEME_DIR=/usr/share/plymouth/themes
+    fi
+    PLYMOUTH_THEME_NAME=${CONFIGURED_THEME_NAME:-spinner}
+}
+
 # called by dracut
 check() {
     [[ "$mount_needs" ]] && return 1
     [[ $(pkglib_dir) ]] || return 1
 
     require_binaries plymouthd plymouth plymouth-set-default-theme || return 1
```

Comment:
```suggestion
    require_binaries plymouthd plymouth || return 1
```

plymouth-set-default-theme is not used any more.